Jenkinsfile

```diff
@@ -0,0 +1,158 @@
+            }
+        }
+    }
+
+    parallel checkouts
+}
+
+if (launchUnitTests.equals("yes")) {
+    stage("Unit tests and Code style") {
+        def tasks = [:]
+
+        tasks["php-cs-fixer"] = {runPhpCsFixerTest()}
+
+        for (phpVersion in supportedPhpVersions) {
+            tasks["phpspec-${phpVersion}"] = {runPhpSpecTest(phpVersion)}
+        }
+
+        parallel tasks
+    }
+}
+
+if (launchIntegrationTests.equals("yes")) {
+    stage("Integration tests") {
+        def tasks = [:]
+
+        for (client in clients) {
+            for (phpVersion in supportedPhpVersions) {
+                tasks["phpunit-${phpVersion}"-${client}] = {runIntegrationTest(phpVersion, client)}
+            }
+        }
+
+        // parallel tasks
+    }
+}
+
+def runPhpCsFixerTest() {
+    node('docker') {
+        deleteDir()
+        try {
+            docker.image("carcel/php:7.1") {
+                unstash "php-api-client_php-http-guzzle6-adapter_php-7.1"
+
+                sh "./bin/php-cs-fixer fix --diff --dry-run --format=junit --config=.php_cs.php > build/logs/phpcs.xml"
+            }
+        } finally {
+            sh "docker stop \$(docker ps -a -q) || true"
+            sh "docker rm \$(docker ps -a -q) || true"
+            sh "docker volume rm \$(docker volume ls -q) || true"
+
+            sh "find build/logs/ -name \"*.xml\" | xargs sed -i \"s/testcase name=\\\"/testcase name=\\\"[php-cs-fixer] /\""
+            junit "build/logs/*.xml"
+
+            deleteDir()
+        }
+    }
+}
+
+def runPhpSpecTest(phpVersion) {
+    node('docker') {
+        deleteDir()
+        try {
+            docker.image("carcel/php:${phpVersion}") {
+                unstash "php-api-client_php-http-guzzle6-adapter_php-${phpVersion}"
+
+                sh "./bin/phpspec run --no-interaction --format=junit > build/logs/phpspec.xml"
+            }
+        } finally {
+            sh "docker stop \$(docker ps -a -q) || true"
+            sh "docker rm \$(docker ps -a -q) || true"
+            sh "docker volume rm \$(docker volume ls -q) || true"
+
+            sh "find build/logs/ -name \"*.xml\" | xargs sed -i \"s/testcase name=\\\"/testcase name=\\\"[php-${phpVersion}] /\""
+            junit "build/logs/*.xml"
+
+            deleteDir()
+        }
+    }
+}
+
+def runIntegrationTest(phpVersion, client) {
+    node('docker') {
+        deleteDir()
+        try {
+            docker.image("carcel/php:${phpVersion}") {
+                unstash "php-api-client_${client.replaceAll('/', '-')}_php-${phpVersion}"
+
+                sh "./bin/phpunit -c app/phpunit.xml.dist --log-junit build/logs/phpunit_integration.xml"
+            }
+        } finally {
+            sh "docker stop \$(docker ps -a -q) || true"
+            sh "docker rm \$(docker ps -a -q) || true"
+            sh "docker volume rm \$(docker volume ls -q) || true"
+
+            sh "sed -i \"s/testcase name=\\\"/testcase name=\\\"[php-${phpVersion}-${client}] /\" build/logs/*.xml"
```

Comment:
you can use xargs as on the pim. Should not be a problem, but just for consistency between the PIM and the client.